shared/createHelper.js

```diff
@@ -163,18 +179,35 @@ function printNextSteps(devToolName, projectPath, result) {
 // Helper for 'create' command
 //
 function createApp(config, platform, devToolName) {
-    var isNative = config.apptype.indexOf('native') >= 0;
-    
+
+    // Computing projectDir
+    config.projectDir = config.outputdir ? path.resolve(config.outputdir) : path.join(process.cwd(),config.appname)
+    config.projectPath = path.relative(process.cwd(), config.projectDir);
+
+    // Checking if directory exists
+    utils.failIfExists(config.projectDir);
+
     // Adding platform
     config.platform = platform;
     
-    // Adding defaults
+    // Adding template repo url and path if none provided
     config.templaterepourl = config.templaterepourl || SDK.templates.repoUrl;
     config.templatepath = config.templatepath || (config.templaterepourl.indexOf('SalesforceMobileSDK-Templates') >= 0 ? SDK.templates.appTypesToPath[platform][config.apptype] : '');
 
-    // Computing projectDir
-    config.projectDir = config.outputdir ? path.resolve(config.outputdir) : path.join(process.cwd(),config.appname)
-    config.projectPath = path.relative(process.cwd(), config.projectDir);
+    // Creating tmp dir for template clone
+    var tmpDir = utils.mkTmpDir();
+
+    // Cloning template repo
+    var repoDir = utils.cloneRepo(tmpDir, config.templaterepourl);
```

Comment:
Getting the template before calling createNativeApp or createHybridApp because that's where we can get the app type.